sdk/python/kfp/compiler/test_data/README.md

```diff
@@ -0,0 +1,7 @@
+# Pipeline samples for compiler unit tests.
+
+To update all golden snapshots:
+
+```bash
+for f in test_data/*.py ; do python3 "$f" ; done
```

Comment:
https://github.com/kubeflow/pipelines/pull/7649 is a conflicting PR with this one that adds a golden snapshot generation CLI, since this shell script does not handle component compilation. I'm fine with either being merged first, but the other will need to be updated when the first is merged.

Reply:
I see, let me keep it for now and comment in the other PR.